lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java

```diff
@@ -51,7 +59,31 @@ public static int compareStrings(String a, String b) {
    * @return The letter grade ("A+", "A", "A-", "B+", etc.).
    */
   public static String convertGpaToLetterGrade(double gpa) {
-    return "F";
+    if (gpa == 4.0) {
+      return "A";
+    } else if (gpa >= 3.7 && gpa < 4.0) {
+      return "A-";
+    } else if (gpa >= 3.3 && gpa < 3.7) {
+      return "B+";
+    } else if (gpa >= 3.0 && gpa < 3.3) {
+      return "B";
+    } else if (gpa >= 2.7 && gpa < 3.0) {
+      return "B-";
+    } else if (gpa >= 2.3 && gpa < 2.7) {
+      return "C+";
+    } else if (gpa >= 2.0 && gpa < 2.3) {
+      return "C";
+    } else if (gpa >= 1.7 && gpa < 2.0) {
+      return "C-";
+    } else if (gpa >= 1.3 && gpa < 1.7) {
+      return "D+";
+    } else if (gpa >= 1.0 && gpa < 1.3) {
+      return "D";
+    } else if (gpa >= 0 && gpa < 1.0) {
+      return "F";
+    } else {
+      return "Error: GPA entered is not valid";
```

Comment:
How would one distinguish between this and any other valid letter grade? Using magic values like this is an anti-pattern. It would be best to throw an exception here.